openedx/core/djangoapps/content_libraries/rest_api/blocks.py

```diff
@@ -60,7 +61,9 @@ def get(self, request, lib_key_str):
         text_search = request.query_params.get('text_search', None)
         block_types = request.query_params.getlist('block_type') or None
 
-        api.require_permission_for_library_key(key, request.user, permissions.CAN_VIEW_THIS_CONTENT_LIBRARY)
+        is_allowed_with_authz = authz_api.is_user_allowed(request.user.username, 'view_library', str(lib_key_str))
+        if not is_allowed_with_authz:
+            api.require_permission_for_library_key(key, request.user, permissions.CAN_VIEW_THIS_CONTENT_LIBRARY)
```

Comment:
Why not, instead of modifying all these entrypoints, we only modify the `require_permission_for_library_key` method - after some tweaks, so it handles both authorization paths?

Reply:
Does this mean that we're checking twice in some cases?